### ISSUE TITLE
How to Speed up Inference?

### ISSUE THREAD

**Issue by admin0319** [Ixiaohuihuihui/Tiny-Defect-Detection-for-PCB#7]
Hi. 
Thank you so much for your kind reply.

Now, We are demonstrating through your [Tiny-Defect-Detection-for-PCB] code like the following video.(https://youtu.be/w7wa3DGBt9A)

We are very satisfied PCB defect detection accuraccy!

However, it takes 6~9 seconds per case, so I want to save time.

1.
There are 6 bad classes and we trained through resnet V1 50. We train a only 100 images, 
320mb weight file created.

I think reducing the weight file size will shorten the inference time. What changes do you make to cfgy.py to reduce it?

2. 
My input image file is grayscale,1280x960 as shown in the picture. 

![pcb00091](https://user-images.githubusercontent.com/35323888/63634058-44975400-c68c-11e9-8858-6b4343c6062f.JPG)

Is there any code to modify to optimize the model for this input image?

3.
Is it more useful to learn using mobilenet than resnet?

Please answer.
Thank you very much.

**Comment by Ixiaohuihuihui** [comment]
Hi,
I'm very happy to see this video demo!
This is a good question, I also think the inference speed is slow and need to speed up.  

1. I save all the weights、biases、gradients parameters and graph, so the weight files are a little big. Actually, you can specify the parameter of tf.saver.
For example:
If you want to save the parameters after 8000  or 10000 iteration steps, the code can be writen as:
[https://github.com/Ixiaohuihuihui/Tiny-Defect-Detection-for-PCB/blob/f59e431bd10f3d9882308f5c2fb5c825937668ce/tools/train.py#L182](url)
` saver.save(sess, save_ckpt， global_step=8000)`
If you don't want to save graph, the code can be writen as:
` saver.save(sess, save_ckpt， global_step=8000, write_meta_graph=False)`
In all, maybe you should understand the tensoflow Save and Restore[https://www.tensorflow.org/guide/saved_model](url)

2. Nope, Is your picture single channel? Because color image is 3 channel (RGB), I think you can modify the channel of convolution layer (I'm not sure). And you can change PIXEL_MEAN in cfg.py, because it is 3 channels.

3. I didn't try mobilenet, but I think it will be helpful in speeding up inference. 
Hope you can get a good result!

**Comment by Ixiaohuihuihui** [comment]
In addition, you don't need to train 30000 iterations, In order to balance accuracy and speed, the number of iterations can be reduced to 10,000.

**Comment by admin0319** [comment]
Thank you for your kind reply.

The methods you told me couldn't reduce the file size. 
I found that reducing the ROI_SIZE in config makes the weight file smaller. 
However, the inference speed did not differ much.
